### PR TITLE
Fix exception when submitting invalid repo params #314

### PR DESCRIPTION
**Pull request by chrisccerami**
Rather than raising an exception when the user enters invalid params to find a repo, it should now render the form with appropriate errors.

### CODE REVIEW

**Comment by prathamesh-sonpatki** [comment]
@chrisccerami What was the error that we were getting with invalid params?

**Comment by chrisccerami** [comment]
If a user click "Add Repo" without filling in the form, the site would crash. The same would happen if the user entered something that wasn't a real repo, like if they had a typo in their entry. Now it correctly renders the /repos/new page again with error messages.

File: app/controllers/repos_controller.rb [hunk 1]
```diff
@@ -85,8 +83,12 @@ def parse_params_for_repo_info
       if params[:url]
         params[:url].match(/^https:\/\/github\.com\/([^\/]*)\/([^\/]*)\/?$/)
         params[:repo] ||= {}
-        params[:repo][:user_name] = $1
-        params[:repo][:name] = $2
+        params[:repo][:user_name] = $1.to_s
+        params[:repo][:name] = $2.to_s
```
Comment (schneems) [inline app/controllers/repos_controller.rb]: Didn't realize this match logic was in here. We don't need to do it now, but we should eventually switch to a match block and named matches

```ruby
params[:repo] ||= {}
params[:url].match(/^https:\/\/github\.com\/(?<user_name>[^\/]*)\/(?<name>[^\/]*)\/?$/) do |m|
  params[:repo][:user_name] = m[:user_name]
  params[:repo][:name]      = m[:name]
end
```

File: app/controllers/repos_controller.rb [hunk 2]
```diff
@@ -85,8 +83,12 @@ def parse_params_for_repo_info
...
       end
     end
+
+    def params_blank?
+      repo_params.values.any? { |param| param.blank? }
```
Comment (schneems) [inline app/controllers/repos_controller.rb]: You can shorten this up

```ruby
repo_params.values.any?(&:blank?)
```

File: app/views/shared/_add_repos_list.html.slim [hunk 3]
```diff
@@ -1,4 +1,4 @@
-- if repos.size == 0
+- if repos.blank? || repos.size == 0
```
Comment (schneems) [inline app/views/shared/_add_repos_list.html.slim]: The `blank?` method will cover when an array is empty.

```
[].blank?
=> true
```

We could remove the `repos.size == 0 ` check.

**Comment by schneems** [comment]
I'm a big fan of nice helpful error messages. Thanks! :heart: I made some style comments that aren't crucial to execution but would be nice to have. If you're interested in another PR, could you clean up some of the items here? Welcome to the codetriage contributor list :rocket:

**Comment by chrisccerami** [comment]
Yeah I can clean up those comments in a few hours. Thanks for the feedback.

**Comment by chrisccerami** [comment]
Ok I fixed the last two comments. I also realized the site crashes if you try to submit a repo that's already been used. Let me know if you'd like me to address that in this PR or if I should open another for it.

**Comment by schneems** [comment]
Can you open a new PR for the new issue please? Thanks for your help

**Comment by schneems** [comment]
This is still failing ` undefined method `errors' for nil:NilClass`

**Pull request merged**